tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -761,6 +823,38 @@ impl Style {
     }
 }
 
+struct FmtThreadName<'a> {
+    name: &'a str,
+}
+
+impl<'a> FmtThreadName<'a> {
+    pub(crate) fn new(name: &'a str) -> Self {
+        Self { name }
+    }
+}
+
+impl<'a> fmt::Display for FmtThreadName<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use std::sync::atomic::{
+            AtomicUsize,
+            Ordering::{AcqRel, Acquire, Relaxed},
+        };
+
+        static MAX_LEN: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Nit: can you add a comment explaining why this is necessary? If I'm following along correctly, this sets the maximum length a thread name can be, to ensure better formatted logs? If so, can you add a comment saying that?

Also, what is the _scope_ of `MAX_LEN`? It doesn't get dropped once this `fmt` call ends, does it?

Reply:
> Also, what is the _scope_ of `MAX_LEN`? It doesn't get dropped once this `fmt` call ends, does it?

Statics don't get dropped. The only difference between a static inside a function and a static outside of a function is visibility; the function-local static is only visible to that function.

Reply:
re:
> Nit: can you add a comment explaining why this is necessary?

@Pothulapati, IIRC, when I suggested we do this, I think I sent a version of this code with comments explaining what it was doing; if you like, please feel free to reuse that.

Reply:
done! :)